### ISSUE TITLE
[FR] Allow uploading .zip files

### ISSUE THREAD

**Issue by sgmills** [grahampugh/jamf-upload#87]
Jamf distribution points are capable of accepting .zip files that contain a .pkg (and optionally other config files). The Jamf agent will then unzip the file and run the pkg installer contained within. This is useful when certain installers require some configuration file or other data to be present in the same directory during installation.

I am requesting that JamfPackageUploader support this workflow, if possible.

**Comment by grahampugh** [comment]
Do you have any documentation for this? I've always put everything required into a flat package myself.

**Comment by sgmills** [comment]
I have never really come across any documentation from the Jamf side, this is just something that works (perhaps unofficially?). 

I have encountered a couple apps where this method works well. The F5 Big-IP Edge Client is one. If you zip the installer package with the configuration file `config_tmp.f5c` then jamf will install and configure it correctly. Another app that works like this is Palo Alto Cortex XDR. Their docs actually say to "Upload the ZIP package you downloaded from Cortex XDR to your MDM. Do not extract it." [Source](https://docs-cortex.paloaltonetworks.com/r/Cortex-XDR/8.3/Cortex-XDR-Agent-Administrator-Guide/Install-the-Cortex-XDR-Agent-for-Mac)

I know that some adobe installers used to run from zip files as well, but I don't deploy those any longer so I'm not sure how things are working today.

**Comment by grahampugh** [comment]
Yes, Adobe CC apps are still bundle-style pkgs and they need to be zipped to upload to Jamf. (The GUI, Jamf Admin and JamfUploader handle that). I just had no idea that you could put extra files in the archive. Did you ever try putting more than one package in the zip?

**Comment by sgmills** [comment]
Multiple packages in one zip is not something I have ever tried.

**Comment by grahampugh** [comment]
I just tried it, and only one of the packages in the zip was installed (alphabetically it was the second package, so who knows how it chooses which one). So I guess the content beyond the package is only useful if the installer package is looking for it.

Still, no reason not to allow JamfUploader to deal with them. I'll add it to the list (well, Issues is the list, so it's already here!).

**Comment by grahampugh** [comment]
Hi @sgmills, I just uploaded a zip file containing a .pkg and a txt file using JamfPackageUploader. It uploaded just fine. Have you encountered any problems when you tried it?

**Comment by sgmills** [comment]
I hadn't tried again since submitting this. Just did a test and this time I was able to upload a .zip with JamfPackageUploader

**Comment by grahampugh** [comment]
Nice, easiest fix ever - it just works 😀
